src/test/java/mate/academy/springbootintro/controllers/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,184 @@
+package mate.academy.springbootintro.controllers;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.when;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.sql.DataSource;
+import lombok.SneakyThrows;
+import mate.academy.springbootintro.dto.cartitem.CartItemDto;
+import mate.academy.springbootintro.dto.cartitem.CreateCartItemRequestDto;
+import mate.academy.springbootintro.dto.cartitem.UpdateCartItemRequestDto;
+import mate.academy.springbootintro.dto.shoppingcartdto.ShoppingCartDto;
+import mate.academy.springbootintro.service.shoppingcart.ShoppingCartService;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithUserDetails;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.context.WebApplicationContext;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@Transactional
```

Comment:
Why do you need this annotation here?

Reply:
Because tests are not passed without annotation @Transactional

Reply:
I'm not sure that the problems are related to this annotation
remove it